impl/client/src/main/javascript/web/util/_a11y.js

```diff
@@ -35,6 +36,385 @@ define(["./_focus"], function(focusUtil) {
+     *
+     * @param {KeyboardEvent} evt - The keydown event object.
+     */
+    function onKeyPress(evt) {
+      var key = evt.key;
+      var lastActiveId = activeDescendantId;
+      var currentItem = getActiveOption() || getAllOptions()[0];
+      var nextItem = currentItem;
+
+      if (!currentItem) {
+        return;
+      }
+
+      switch (key) {
+        case keys.PageUp:
+        case keys.PageDown:
+          break;
+        case keys.ArrowUp:
+        case keys.ArrowDown:
+          evt.preventDefault();
+          if (!activeDescendantId) {
+            // focus first option if no option was previously focused, and perform no other actions
+            setActiveOption(currentItem);
+            break;
+          }
+
+          if (key === keys.ArrowUp) {
+            nextItem = findPreviousOption(currentItem);
+          } else {
+            nextItem = findNextOption(currentItem);
+          }
+
+          if (nextItem && event.shiftKey) {
+            selectRange(startRangeIndex, nextItem);
+          }
+
+          if (nextItem) {
+            setActiveOption(nextItem);
+            evt.preventDefault();
+          }
+
+          break;
+        case keys.Home:
+          evt.preventDefault();
+          setActiveFirstOption();
+
+          if (evt.shiftKey && evt.ctrlKey) {
+            selectRange(startRangeIndex, 0);
+          }
+          break;
+        case keys.End:
+          evt.preventDefault();
+          setActiveLastOption();
+
+          if (evt.shiftKey && evt.ctrlKey) {
+            selectRange(startRangeIndex, getAllOptions().length - 1);
+          }
+          break;
+        case keys.Shift:
+          startRangeIndex = getOptionIndex(currentItem);
+          break;
+        case keys.Space:
+          evt.preventDefault();
+          toggleSelected(nextItem);
+          break;
+        case keys.a:
+        case keys.A:
+          // handle control + A
+          if (evt.ctrlKey || evt.metaKey) {
+            evt.preventDefault();
+            selectRange(0, getAllOptions().length - 1);
+            break;
+          }
+          // fall through for Type-ahead
+        default:
+          var character = key.toUpperCase();
+          var startIndex = !keysSoFar ? getActiveOptionIndex() : null;
+
+          keysSoFar += character;
+
+          clearKeysSoFarAfterDelay();
+
+          var optionToFocus = findOption(keysSoFar, startIndex);
+          if (optionToFocus) {
+            setActiveOption(optionToFocus);
+          }
+          break;
+      }
+
+      if (activeDescendantId !== lastActiveId) {
+        scrollActiveOptionIntoView();
+      }
+    }
+
+    /**
+     * Helper function to check if a number is within a range; no side effects.
+     *
+     * @param index - option index to be searched for.
+     * @param start - search starts at this option index.
+     * @param end - search ends at this option index.
+     * @return {boolean} - true if the number falls within the range else false.
```

Comment:
```js
@param {number} index - ...
```

Same for other parameters.

Also, for the `@return` tag, no separator `-` is/can be used.